src/RescriptEmbedLang.res

```diff
@@ -166,6 +174,24 @@ type handleOtherCommandConfig<'config> = {
 
 type setupConfig = {args: CliArgs.t}
 
+type watcherOnChangeConfig = {
+  file: string,
+  runGeneration: (~files: array<string>=?) => promise<unit>,
+}
+
+type watcher = {
+  filePattern: string,
+  onChange: watcherOnChangeConfig => promise<unit>
+}
+
+@unboxed
+type setupResult<'config> =
+  | SetupResult({
+      config: 'config,
+      additionalFileWatchers?: array<watcher>,
+      additionalIgnorePatterns?: array<string>
+    })
```

Comment:
Initially used a plain-ol' record here, but that caused some inference issues due to the `config` parameter, so I swapped it out for the unboxed single-case variant to sidestep that issue.